### PR TITLE
Feature/32/consensus info via grpc

### PR DESCRIPTION
**Pull request by blazejkrzak**
This is upgrade for GRPC and removal of RPC support in vanguard repository
#32

### CODE REVIEW

**Comment by blazejkrzak** [comment]
I am awaiting for minimalConsensusInfo validator list (was missing after @atif-konasl build of vanguard-apis)

**Review by atif-konasl — COMMENTED** [review]
@blazejkrzak looks good to me. We will add more test cases when doing refactor.

File: orchestrator/vanguardchain/service.go [hunk 1]
```diff
@@ -223,46 +197,27 @@ func (s *Service) retryVanguardNode(err error) {
 }
 
 func (s *Service) subscribeToVanguardGRPC() (err error) {
-	// subscribe to vanguard client for new pending blocks
-	// TODO: add this client into dependency injection
-	// Vanguard endpoint will be invalid I guess as long as we support both grpc and rpc
 	vanguardClient, err := s.dialGRPCFn(s.vanGRPCEndpoint)
 
 	if nil != err {
```
Comment (atif-konasl) [inline orchestrator/vanguardchain/service.go:202]: @blazejkrzak it's may be unintentional. Should return error

File: orchestrator/vanguardchain/client/client.go [hunk 2]
```diff
@@ -102,6 +96,26 @@ func (vanClient *GRPCClient) StreamNewPendingBlocks() (
...
+	stream ethpb.BeaconChain_StreamMinimalConsensusInfoClient,
+	err error,
+) {
+	stream, err = vanClient.beaconClient.StreamMinimalConsensusInfo(
+		vanClient.ctx,
+		&ptypes.Empty{},
+	)
+
+	if err != nil {
+		log.WithError(err).Fatal("Failed to subscribe to StreamMinimalConsensusInfo")
```
Comment (atif-konasl) [inline orchestrator/vanguardchain/client/client.go:109]: @blazejkrzak application stops if it fails to subscribe. But in practical scenario, node should not shutdown. It should re-try again and again.

**Review by atif-konasl — COMMENTED** [review]
@blazejkrzak Without vanguard node, the orchestrator node automatically gets shut down instead of retrying. Please fix this.

File: orchestrator/vanguardchain/sevice_test.go [hunk 3]
```diff
@@ -12,64 +14,36 @@ import (
...
 	assert.LogsContain(t, hook, "Connected vanguard chain")
 	assert.LogsContain(t, hook, "subscribed to vanguard chain for consensus info")
 	hook.Reset()
-	assert.NoError(t, vanSvc.Stop())
-
 }
 
-// Test_VanguardSvc_NoServerConn checks that vanguard service
-// should try to ping the server after certain period
-func Test_VanguardSvc_NoServerConn(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (blazejkrzak) [reply orchestrator/vanguardchain/sevice_test.go:37]: I have no intention of mocking retries on grpc

[the remark this answers is not in the document]
Comment (blazejkrzak) [reply orchestrator/vanguardchain/sevice_test.go:37]: I would move this test to higher level.
It was inproc done by rpc, but atm I think its better to not implement fake connection drop

[the remark this answers is not in the document]
Comment (blazejkrzak) [reply orchestrator/vanguardchain/sevice_test.go:37]: Let me think about it

File: orchestrator/vanguardchain/sevice_test.go [hunk 4]
```diff
@@ -12,64 +14,36 @@ import (
...
-
-	time.Sleep(2 * time.Second)
-	assert.LogsContain(t, hook, "Could not connect to vanguard endpoint")
-	hook.Reset()
+func CleanConsensusMocks() {
+	ConsensusInfoMocks = nil
 }
 
-// Test_VanguardSvc_RetryToConnServer
-func Test_VanguardSvc_RetryToConnServer(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (blazejkrzak) [reply orchestrator/vanguardchain/sevice_test.go:53]: see above

File: orchestrator/vanguardchain/client/client.go (hunk 2 above)
Comment (blazejkrzak) [reply orchestrator/vanguardchain/client/client.go:109]: could not be fatal, nice catch

File: orchestrator/rpc/service_test.go [hunk 5]
```diff
@@ -14,23 +13,14 @@ import (
...
-	}
 
 	namespace := "van"
 	consensusInfoFeed, err := vanguardchain.NewService(
 		context.Background(),
-		cmd.DefaultVanguardRPCEndpoint,
 		cmd.DefaultVanguardGRPCEndpoint,
 		namespace,
 		orchestratorDB,
 		orchestratorDB,
```
[the remark this answers is not in the document]
Comment (blazejkrzak) [reply orchestrator/rpc/service_test.go:23]:
```yaml
    steps:
      - name: Install Go
        uses: actions/setup-go@v2
        with:
          go-version: ${{ matrix.go-version }}

      - name: Checkout code
        uses: actions/checkout@v2

      - name: Test
        run: go test ./...
```

[the remark this answers is not in the document]
Comment (blazejkrzak) [reply orchestrator/rpc/service_test.go:23]: https://github.com/lukso-network/lukso-orchestrator/pull/33/checks?check_run_id=2618543213

[the remark this answers is not in the document]
Comment (blazejkrzak) [reply orchestrator/rpc/service_test.go:23]: all test pass for this commit

**Comment by blazejkrzak** [comment]
@atif-konasl  pushed some refactor of NewService and resolved fatal log

File: orchestrator/vanguardchain/sevice_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (blazejkrzak) [reply orchestrator/vanguardchain/sevice_test.go:37]: Ok, it was not so hard, I've restored them

File: orchestrator/vanguardchain/sevice_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (blazejkrzak) [reply orchestrator/vanguardchain/sevice_test.go:53]: Ok, it was not so hard, I've restored them

**Comment by blazejkrzak** [comment]
It passes, works and all the comments were solved. I think that its ready to merge

**Review by chain-oracle — APPROVED** [review]

**Review by atif-konasl — APPROVED** [review]

**Pull request merged**